crates/macros/src/lib.rs

```diff
@@ -0,0 +1,53 @@
+extern crate proc_macro;
+
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{self, Data, DeriveInput, Fields};
+
+#[proc_macro_derive(BlockData)]
+pub fn block_data_derive(input: TokenStream) -> TokenStream {
+    let ast = syn::parse(input).expect("Parsing struct for BlockData macro failed");
+
+    impl_block_data(&ast)
+}
+
+fn impl_block_data(ast: &DeriveInput) -> TokenStream {
+    let name = &ast.ident;
+    let field;
+
+    if let Data::Struct(data) = &ast.data {
+        if let Fields::Named(named_fields) = &data.fields {
+            if named_fields.named.len() != 1 {
+                panic!("Can't derive BlockData for struct with more than 1 named field")
```

Comment:
Errors in procmacro should be handled with or similar
https://docs.rs/syn/1.0.60/syn/struct.Error.html#method.new

Reply:
Thanks for pointing that out!